04_snake_game/main.go

```diff
@@ -98,8 +150,24 @@ func moveSnake(s snake, v coord, fw, fh int) snake {
 	return s
 }
 
+func collisions(g game) {
+	if g.sn.pos.x == 0 || g.sn.pos.x == g.fieldWidth-1 || g.sn.pos.y == 1 || g.sn.pos.y == g.fieldHeight-1 {
+		writeText(g.fieldWidth/2, g.fieldHeight/2, "Game Over", termbox.ColorRed|termbox.AttrBold, termbox.ColorDefault)
+		termbox.Flush()
+		time.Sleep(5 * time.Second)
+		termbox.Clear(snakeFgColor, snakeBgColor)
+		termbox.Flush()
+		os.Exit(0)
```

Comment:
It's better to terminate the program by returning a flag from the function that indicates if the program needs to be termiated, i.e.

```
func collisions(g game) (isTerminated bool) {
```